vassal-app/src/main/java/VASSAL/build/module/gamepieceimage/GamePieceImage.java

```diff
@@ -463,4 +501,167 @@ protected void rebuildInstances() {
+    public PngImageNameConfigurer(String key, String name, GamePieceImage gpi) {
+      super(key, name, gpi.getConfigureName());
+      this.gpi = gpi;
+      setHintKey("Editor.GamePieceImage.png_image_name");
+      getControls();
+      ((AbstractDocument) nameField.getDocument()).setDocumentFilter(new ImageNameFilter(this));
+    }
+
+    public boolean isGpiVersion1() {
+      return gpi.isVersion1();
+    }
+
+    public void setCaretPosition(int pos) {
+      nameField.setCaretPosition(pos);
+    }
+  }
+
+  /**
+   * ImageNameFilter that controls how the user can change the Image Name
+   * If the GPI is still version 0, then no controls. Not we can't force an image
+   * name because other components may reference that image.
+   * Once the GPI is version 0, then enfore a .png suffix and
+   */
+  private static class ImageNameFilter extends DocumentFilter {
+    private final PngImageNameConfigurer config;
+
+    public ImageNameFilter(PngImageNameConfigurer config) {
+      this.config = config;
+    }
+
+    /** Return the current value of the text */
+    private String getText(FilterBypass fb) {
+      final Document doc = fb.getDocument();
+      if (doc == null || doc.getLength() == 0) {
+        return "";
+      }
+
+      String currentValue;
+      try {
+        currentValue = doc.getText(0, doc.getLength());
+      }
+      catch (BadLocationException ignored) {
+        currentValue = "";
+      }
+
+      return currentValue;
+    }
+
+    /** Does the current text end in '.png'? */
+    private boolean isPng(FilterBypass fb) {
+      return getText(fb).toUpperCase().endsWith(PNG_SUFFIX.toUpperCase());
+    }
+
+    /** Ensure the current value ends in '.png' */
+    private void fixPng(FilterBypass fb) throws BadLocationException {
+      fixPng(fb, 1);
+    }
+
+    private void fixPng(FilterBypass fb, int pos) throws BadLocationException {
+      if (!isPng(fb)) {
+        final String text = getText(fb);
+        if (text.endsWith(".")) {
+          super.replace(fb, text.length() - 1, 1, PNG_SUFFIX, null);
+          config.setCaretPosition(pos);
+        }
+        else if (text.toUpperCase().endsWith(".p")) {
+          super.replace(fb, text.length() - 2, 2, PNG_SUFFIX, null);
+          config.setCaretPosition(pos);
+        }
+        else if (text.toUpperCase().endsWith(".pn")) {
+          super.replace(fb, text.length() - 3, 3, PNG_SUFFIX, null);
+          config.setCaretPosition(pos);
+        }
+        else {
+          super.replace(fb, text.length(), 0, PNG_SUFFIX, null);
+          config.setCaretPosition(pos);
+        }
+      }
+
+    }
+
+    private String clean(String string) {
+      final StringBuilder sb = new StringBuilder();
+      for (int i = 0; i < string.length(); i++) {
+        final char c = string.charAt(i);
+        if (c == '_' || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')) {
+          sb.append(c);
+        }
+      }
+      return sb.toString();
```

Comment:
Could you replace this whole function with `string.replaceAll("\W", "");`?

Reply:
Better version.